packages/crud/src/vaadin-crud.js

```diff
@@ -338,34 +337,31 @@ class Crud extends SlotMixin(ControllerMixin(ElementMixin(ThemableMixin(PolymerE
        * @private
        */
       _saveButton: {
-        type: HTMLElement,
-        observer: '__saveButtonChanged',
+        type: Object,
       },
 
       /**
-       * A reference to the save button which will be teleported to the dialog
+       * A reference to the delete button which will be teleported to the dialog
        * @private
        */
       _deleteButton: {
-        type: HTMLElement,
-        observer: '__deleteButtonChanged',
+        type: Object,
       },
 
       /**
-       * A reference to the save button which will be teleported to the dialog
+       * A reference to the cancel button which will be teleported to the dialog
        * @private
        */
       _cancelButton: {
-        type: HTMLElement,
-        observer: '__cancelButtonChanged',
+        type: Object,
       },
 
       /**
-       * A reference to the editor header element will be teleported to the dialog.
+       * A reference to the default editor header element created by the CRUD
        * @private
        */
-      _headerNode: {
-        type: HTMLElement,
+      _defaultHeader: {
```

Comment:
Changed to only modify text content for default header node, and not update a custom header.
This would be aligned with e.g. how `LabelMixin` works in terms of a custom `<label>`.

Note, custom buttons are still updated because that's how they are expected to work.

Reply:
Arguable, but I think I agree. The alternative would be to listen to events and update the custom header manually.